packages/lending/src/adapter.ts

```diff
@@ -185,6 +185,299 @@ export class Adapter extends common.Web3Toolkit {
+      if (!srcCollateral) throw new OperationError('collateralToken', 'UNSUPPORTED_TOKEN');
+
+      const destBorrow = portfolio.findBorrow(debtToken);
+      if (!destBorrow) throw new OperationError('debtToken', 'UNSUPPORTED_TOKEN');
+
+      const supplyAmountDelta = new common.TokenAmount(collateralToken.wrapped, collateralAmount).sub(
+        srcCollateral.balance
+      );
+      if (!srcCollateral.validateSupplyCap(supplyAmountDelta.amount)) {
+        throw new OperationError('collateralAmount', 'SUPPLY_CAP_EXCEEDED');
+      }
+
+      // 2. ---------- zap swap ----------
+      let supplyInput = new common.TokenAmount(collateralToken.wrapped, '0');
+      if (zapAmount !== '0' && !zapToken.wrapped.is(collateralToken.wrapped)) {
+        const swapper = this.findSwapper([zapToken, collateralToken]);
+        let swapQuotation: SwapperQuoteFields;
+        try {
+          swapQuotation = await swapper.quote({
+            input: new common.TokenAmount(zapToken, zapAmount),
+            tokenOut: collateralToken.wrapped,
+            slippage,
+          });
+        } catch {
+          throw new OperationError('zapAmount', 'NO_ROUTE_FOUND_OR_PRICE_IMPACT_TOO_HIGH');
+        }
+        const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
+        output.logics.push(swapTokenLogic);
+        // 2-1. the supply amount is the swap quotation output
+        supplyInput = swapQuotation.output;
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      } else if (zapToken.wrapped.is(collateralToken.wrapped)) {
+        supplyInput = new common.TokenAmount(zapToken, zapAmount);
+        output.afterPortfolio.supply(supplyInput.token, supplyInput.amount);
+      }
+
+      // 3. leverage by collateral
+      const leverageAmount = supplyAmountDelta.sub(supplyInput.amount).amount;
+      const leverageOutput = await this.leverageByCollateral({
+        account,
+        portfolio: output.afterPortfolio,
+        srcToken: supplyInput.token,
+        srcAmount: leverageAmount,
+        destToken: debtToken,
+      });
+
+      if (leverageOutput.error) throw leverageOutput.error;
+
+      // 3-1. find supply logic and add zap supply amount
+      if (!collateralToken.wrapped.is(debtToken.wrapped)) {
+        leverageOutput.logics[2].fields.input.amount = supplyInput.add(
+          leverageOutput.logics[2].fields.input.amount
+        ).amount;
+        leverageOutput.logics[2].fields.balanceBps = common.BPS_BASE;
+      } else {
+        leverageOutput.logics[1].fields.input.amount = supplyInput.add(
+          leverageOutput.logics[1].fields.input.amount
+        ).amount;
+        leverageOutput.logics[1].fields.balanceBps = common.BPS_BASE;
+      }
+      output.logics.push(...leverageOutput.logics);
+      output.afterPortfolio = leverageOutput.afterPortfolio;
+      output.destAmount = output.afterPortfolio.findBorrow(debtToken)!.balance;
```

Comment:
I think it's reasonable to return `leverageOutput.afterPortfolio.destAmount` as it represents the debt diff amount.

Reply:
prefer to use the debt diff amount to check what changes if I do this transaction

Reply:
Based on our recent verbal discussion, since the leverage part does not manage existing positions, we can directly use the `destAmount` returned by `leverageOutput`.